### DISCUSSION TITLE
is it possible to get the protect attribute when using UC model

### DISCUSSION THREAD

**Discussion by jiangxiaoqiang** [seleniumbase/SeleniumBase#2299] [Q&A] [answered]
I am trying to use SeleniumBase to do some automatic test with Cloudflare turnstile bot-detect. when I tried to get the cloudflare token like this:

  ```
  def get_token_value(self, local_base_case: BaseCase, retries: int, we: WebElement):
        if retries > 30:
            return
        res_filter = "//input[contains(@name,'cf-turnstile-response')]"
        driver: seleniumbase.undetected.Chrome = local_base_case.driver
        driver.switch_to.default_content()
        wait = WebDriverWait(driver, 30)
        resp_element = wait.until(EC.presence_of_element_located((By.XPATH, res_filter)))
        if resp_element:
            input_element = driver.find_element(value=res_filter, by=By.XPATH)
            cf_token_value = input_element.get_attribute("value")
            if cf_token_value is None:
                self.open_the_form_turnstile_page(sb=local_base_case)
            else:
                print(cf_token_value)
        else:
            sleep(3)
            new_reties = retries + 1
            self.get_token_value(local_base_case=local_base_case, retries=new_reties, we=we)


```

it always get the '' with the value, the value is  a long string, why could not get the vlaue? is it possible to get the value?

**Comment by mdmintz** [comment, marked as answer]
It looks like you're doing some odd things in your script. Maybe you should check out the examples:

### Here are some existing examples that use ``--uc`` mode:
* https://github.com/seleniumbase/SeleniumBase/blob/master/examples/verify_undetected.py
* https://github.com/seleniumbase/SeleniumBase/blob/master/examples/uc_cdp_events.py
* https://github.com/seleniumbase/SeleniumBase/blob/master/examples/raw_uc_mode.py

### Here's a raw driver example:

```python
from seleniumbase import Driver
import time

driver = Driver(uc=True)
driver.get("https://nowsecure.nl/#relax")
time.sleep(6)
driver.quit()
```

### Here are some existing examples that use the ``driver.uc_click(selector)`` method:

* [SeleniumBase/examples/raw_turnstile.py](https://github.com/seleniumbase/SeleniumBase/blob/master/examples/raw_turnstile.py)
* [SeleniumBase/examples/raw_form_turnstile.py](https://github.com/seleniumbase/SeleniumBase/blob/master/examples/raw_form_turnstile.py)

--------

With SeleniumBase, you shouldn't need to use `WebDriverWait` at all... automatic waiting is already included with SB methods.

Here's an example of getting the `data-sitekey` on https://seleniumbase.io/apps/recaptcha with the `driver.get_attribute()` method:

```python
from seleniumbase import Driver

driver = Driver(uc=True)
try:
    driver.get("https://seleniumbase.io/apps/recaptcha")
    sitekey = driver.get_attribute("div.g-recaptcha", "data-sitekey")
    print(sitekey)
finally:
    driver.quit()
```

**Reply by Jonathynlee** [reply]
@mdmintz is it possible to interact with elements directly? Setting the WebElement to a variable had been pretty useful for my test case and when I try to extract data from the WebElement directly, I just get the classic "script timeout" error

Just a small example
``` form_html = form.get_attribute('outerHTML').lower()```

**Reply by mdmintz** [reply]
@Jonathynlee Only with raw Selenium methods if they exist. SeleniumBase methods expect a selector to interact with elements.
